extractor-sdk/indexify_extractor_sdk/downloader.py

```
@@ -70,15 +74,122 @@ def install_dependencies(directory_path):
+        # This is to ensure that the database is always
+        # up-to-date with the latest extractor info.
+        cur.execute(f"""
+            DELETE FROM extractors
+            WHERE id='{id}'
+        """)
+
+    input_params: str = description.input_params if description.input_params else None
+
+    # Convert the lists to JSON strings
+    mime_types = json.dumps(description.input_mime_types)
+    embedding_schemas = serialize_embedding_schemas(description.embedding_schemas)
+    metadata_schemas = json.dumps(description.metadata_schemas)
+
+    # Insert the extractor info into the database
+    cur.execute(f"""
+        INSERT INTO extractors (
+            id, name, description, input_params, input_mime_types, metadata_schemas, embedding_schemas
+        ) VALUES (
+            '{id}', '{description.name}', '{description.description}',
+            '{input_params}', '{mime_types}',
+            '{metadata_schemas}', '{embedding_schemas}'
+        )
+    """)
+
+    conn.commit()
+    conn.close()
 
 def download_extractor(extractor_path):
+    # Create extractor database if not exists
+    create_extractor_db()
+
     console.print("[bold #4AA4F4]Downloading Extractor...[/]")
     extractor_path = extractor_path.removeprefix("hub://")
     fs = fsspec.filesystem("github", org="tensorlakeai", repo="indexify-extractors")
+
+    base_extractor_path = os.path.basename(extractor_path)
     directory_path = os.path.join(
-        os.path.expanduser("~"), ".indexify-extractors", os.path.basename(extractor_path)
+        os.path.expanduser("~"), ".indexify-extractors", base_extractor_path
     )
+
+    # Store the extractor info in the database
+
+    extractor_full_name = get_extractor_full_name(base_extractor_path)
+
+    try:
+        description = get_extractor_description(extractor_full_name)
+        save_extractor_description(extractor_full_name, description)
+    except Exception as e:
+        print(f"Error saving extractor description: {e}")
```

This code suppresses the error, should reraise exception to notify the caller of the failure

---

Added the re-raise!